homeassistant/components/tplink/light.py

```diff
@@ -474,6 +476,26 @@ def _set_device_state(self, state):
 
         return self._get_device_state()
 
+    async def async_update(self):
+        """Update the TP-Link bulb's state."""
+        for update_attempt in range(MAX_ATTEMPTS):
```

Comment:
This can give warnings from core that update is taking more than 10 seconds. Maybe that is desirable, I don't know :). An alternative is to not loop and just set it unavailable if update fails two or three times in a row.